pyomo/gdp/plugins/chull.py

```diff
@@ -840,5 +853,131 @@ def _xform_constraint(self, obj, disjunct, infodict, var_substitute_map,
+
+        Parameters
+        ----------
+        v: a Var which appears in a constraint in a transformed Disjunct
+        disjunct: a transformed Disjunct in which v appears
+        """
+        if disjunct._transformation_block is None:
+            raise GDP_Error("Disjunct %s has not been transformed" 
+                            % disjunct.name)
+        transBlock = disjunct._transformation_block()
+        try:
+            return transBlock._disaggregatedVarMap['disaggregatedVar'][v]
+        except:
+            raise GDP_Error("It does not appear %s is a "
+                            "variable which appears in disjunct %s"
+                            % (v.name, disjunct.name))
+        
+    def get_src_var(self, disaggregated_var):
+        """
+        Returns the original model variable to which disaggregated_var 
+        corresponds.
+
+        Parameters
+        ----------
+        disaggregated_var: a Var which was created by the chull 
+                           transformation as a disaggregated variable 
+                           (and so appears on a transformation block 
+                           of some Disjunct)
+        """
+        transBlock = disaggregated_var.parent_block()
+        try:
+            src_disjunct = transBlock._srcDisjunct()
+        except:
+            raise GDP_Error("%s does not appear to be a disaggregated variable"
+                            % disaggregated_var.name)
+        return transBlock._disaggregatedVarMap['srcVar'][disaggregated_var]
+
+    # retrieves the disaggregation constraint for original_var resulting from
+    # transforming disjunction
+    def get_disaggregation_constraint(self, original_var, disjunction):
+        """
+        Returns the disaggregation (re-aggregation?) constraint 
+        (which links the disaggregated variables to their original) 
+        corresponding to original_var and the transformation of disjunction.
+
+        Parameters
+        ----------
+        original_var: a Var which was disaggregated in the transformation
+                      of Disjunction disjunction
+        disjunction: a transformed Disjunction containing original_var
+        """
+        for disjunct in disjunction.disjuncts:
+            transBlock = disjunct._transformation_block
+            if not transBlock is None:
+                break
+        if transBlock is None:
+            raise GDP_Error("Disjunction %s has not been properly transformed: "
+                            "None of its disjuncts are transformed." 
+                            % disjunction.name)
+        try:
+            return transBlock().parent_block()._disaggregationConstraintMap[
+                original_var][disjunction]
+        except:
+            raise GDP_Error("It doesn't appear that %s is a variable that was "
+                            "disaggregated by Disjunction %s" % 
+                            (original_var.name, disjunction.name))
```

Comment:
This is dangerous as it can hide underlying bugs.  Either catch a specific exception, or (probably better) log this message as an error and re-raise the original exeception.